distribution/lib/Standard/Base/0.0.0-dev/src/Data/Array.enso

```diff
@@ -135,23 +145,138 @@ type Array
     length : Integer
     length self = @Builtin_Method "Array.length"
 
-    ## Sorts the Array.
-
-      Arguments:
-      - comparator: A comparison function that takes two elements and returns
-        an Ordering that describes how the first element is ordered with
-        respect to the second.
-
-      > Example
-        Getting a sorted array of numbers.
-
-            [3,2,1].to_array.sort
-    sort : (Any -> Any -> Ordering) -> Array
-    sort self comparator=(Ordering.compare _ _) =
-        self.sort_builtin comparator
-
     ## Identity.
        This method is implemented purely for completeness with the runtime's
        primitive array protocol.
     to_array : Array
-    to_array self = @Builtin_Method "Array.to_array"
+    to_array self = self
+
+    #
+    # For compatibility with Vector
+    #
+
+    sort : Sort_Direction -> (Any -> Any) -> (Any -> Any -> Ordering) -> Vector Any ! Incomparable_Values
+    sort self (order = Sort_Direction.Ascending) (on = x -> x) (by = (Ordering.compare _ _)) = Vector.sort self order on by
+
+    take : (Index_Sub_Range | Range | Integer) -> Vector Any
+    take self range=(Index_Sub_Range.First 1) = Vector.take self range
+
```

Comment:
These should definitely have some doc comments on them, no?

I imagine we can get hold of an Array in the GUI, if yes we need methods on it to display nicely in the Component Browser. We can probably adapt the comments from the Vector ones.

Reply:
The documentation shall be almost an exact copy of the one from `Vector`. Keeping the documentation consistent with `Vector` methods is going to be an issue.

Reply:
Test that compares documentation of `Vector` and `Array` methods written: [0d040e0](https://github.com/enso-org/enso/pull/6218/commits/0d040e02a9ab369b8562e966ebe4271e581e95bd). It checks that all `Vector` methods have appropriate pair in `Array` and that the documentation is (after few transformations) the same.